src/Bicep.Core/Analyzers/Linter/Rules/BCPL1030.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Interfaces;
+using Bicep.Core.CodeAction;
+using Bicep.Core.Extensions;
+using Bicep.Core.Parsing;
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Bicep.Core.Analyzers.Linter.Rules
+{
+    public sealed class BCPL1030 : LinterRule
+    {
+        public BCPL1030() : base(
+            code: "BCPL1030",
+            ruleName: "Secure parameter default not allowed",
+            description: "Secure parameters can't have hardcoded default. This prevents storage of sensitive data in the Bicep declaration.",
+            docUri: "https://bicep/linter/rules/BCPL1030") // TODO: setup up doc pages
+        { }
+
+        public override IEnumerable<IBicepAnalyzerDiagnostic> Analyze(SemanticModel model)
+        {
+            var defaultValueSyntaxes = model.Root.ParameterDeclarations.Where(p => p.IsSecure())
+                .Select(p => p.Modifier as ParameterDefaultValueSyntax)
+                .OfType<ParameterDefaultValueSyntax>(); // this eliminates nulls
+
+            foreach (var defaultValueSyntax in defaultValueSyntaxes)
+            {
+                yield return CreateFixableDiagnosticForSpan(defaultValueSyntax.Span,
+                    new CodeFix("Remove default value of secure parameter", true,
+                            new CodeReplacement(defaultValueSyntax.Span, string.Empty)));
```

Comment:
>defaultValueSyntax [](http://example.com/codeflow?start=48&length=18)

I don't think this accounts for people specifying a default using the deprecated modifier syntax. Although, I think we shouldn't invest too much into handling that syntax. Probably best option would be to omit the code fix if they are using that syntax.

Reply:
Decision is to not implement to cover deprecated modifier syntax